pico/tools/src/main/java/io/helidon/pico/tools/DefaultInterceptorCreator.java

```diff
@@ -0,0 +1,946 @@
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+import java.util.TreeMap;
+import java.util.TreeSet;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.stream.Collectors;
+
+import javax.annotation.processing.ProcessingEnvironment;
+import javax.lang.model.element.AnnotationMirror;
+import javax.lang.model.element.ElementKind;
+import javax.lang.model.element.ExecutableElement;
+import javax.lang.model.element.Modifier;
+import javax.lang.model.element.TypeElement;
+import javax.lang.model.util.Elements;
+
+import io.helidon.builder.processor.tools.BuilderTypeTools;
+import io.helidon.common.LazyValue;
+import io.helidon.common.Weight;
+import io.helidon.common.Weighted;
+import io.helidon.common.types.AnnotationAndValue;
+import io.helidon.common.types.DefaultTypeName;
+import io.helidon.common.types.TypeName;
+import io.helidon.pico.ElementInfo;
+import io.helidon.pico.InjectionPointInfo;
+import io.helidon.pico.InterceptedTrigger;
+import io.helidon.pico.ServiceInfoBasics;
+import io.helidon.pico.spi.Resetable;
+import io.helidon.pico.tools.spi.InterceptorCreator;
+
+import io.github.classgraph.ClassInfo;
+import io.github.classgraph.MethodInfo;
+import io.github.classgraph.ScanResult;
+import jakarta.inject.Singleton;
+
+import static io.helidon.common.types.DefaultAnnotationAndValue.create;
+import static io.helidon.pico.tools.TypeTools.createAnnotationAndValueFromMirror;
+import static io.helidon.pico.tools.TypeTools.createAnnotationAndValueListFromAnnotations;
+import static io.helidon.pico.tools.TypeTools.createAnnotationAndValueSet;
+import static io.helidon.pico.tools.TypeTools.createMethodElementInfo;
+import static io.helidon.pico.tools.TypeTools.gatherAllAnnotationsUsedOnPublicNonStaticMethods;
+import static io.helidon.pico.tools.TypeTools.toKind;
+import static io.helidon.pico.tools.TypeTools.toObjectTypeName;
+
+/**
+ * The default {@link io.helidon.pico.tools.spi.InterceptorCreator} provider in use.
+ */
+@Singleton
+@Weight(Weighted.DEFAULT_WEIGHT)
+@SuppressWarnings("unchecked")
+public class DefaultInterceptorCreator extends AbstractCreator implements InterceptorCreator, Resetable {
+    private static final LazyValue<ScanResult> SCAN = LazyValue.create(ReflectionHandler.INSTANCE::scan);
+
+    private static final String INTERCEPTOR_NAME_SUFFIX = "Interceptor";
+    private static final String INNER_INTERCEPTOR_CLASS_NAME = "$$" + NAME_PREFIX + INTERCEPTOR_NAME_SUFFIX;
+    private static final String COMPLEX_INTERCEPTOR_HBS = "complex-interceptor.hbs";
+    private static final double INTERCEPTOR_PRIORITY_DELTA = 0.001;
+    private static final String CTOR_ALIAS = "ctor";
+    private static final Set<String> MANUALLY_WHITE_LISTED = new LinkedHashSet<>();
```

Comment:
We don't use whitelist/blacklist any more, use allowlist instead please

Reply:
next push.